Client/Frontend/Settings/IPFSCustomGatewayViewController.swift

```diff
@@ -0,0 +1,283 @@
+      case .loading:
+        self.navigationItem.rightBarButtonItem = UIBarButtonItem(customView: self.spinnerView)
+        self.spinnerView.startAnimating()
+      }
+    }
+  }
+  
+  private func showWrongGatewayUrlAlert() {
+    let alert = UIAlertController(
+      title: Strings.BraveIPFS.wrongGatewayAlertTitle,
+      message: Strings.BraveIPFS.wrongGatewayAlertDescription,
+      preferredStyle: .alert
+    )
+    
+    let okayOption = UIAlertAction(
+      title: Strings.OKString,
+      style: .default,
+      handler: nil
+    )
+    
+    alert.addAction(okayOption)
+    present(alert, animated: true) {
+      self.changeAddButton(for: .disabled)
+    }
+  }
+  
+  @objc func validateGateway() {
+    guard let urlTextValue = urlText else {
+      showWrongGatewayUrlAlert()
+      return
+    }
+    if let url = URL(string: urlTextValue) {
+      if !url.isWebPage() {
+        showWrongGatewayUrlAlert()
+        return
+      }
+      var checkUrl = url
+      checkUrl.append(pathComponents: IPFSCustomGatewayViewController.ipfsTestPath, IPFSCustomGatewayViewController.ipfsTestCID)
+      
+      changeAddButton(for: .loading)
+      view.endEditing(true)
+      
+      NetworkManager().downloadResource(with: checkUrl) { [weak self] response in
+        guard let self = self else { return }
+        switch response {
+        case .success(let response):
+          if (String(data: response.data, encoding: .utf8) == IPFSCustomGatewayViewController.ipfsTestContent) {
+            self.setGateway(url)
+          } else {
+            self.showWrongGatewayUrlAlert()
+          }
+        case .failure(_):
+          self.showWrongGatewayUrlAlert()
+        }
+        self.changeAddButton(for: .disabled)
+      }
+    } else {
+      showWrongGatewayUrlAlert()
+      self.changeAddButton(for: .disabled)
+    }
+  }
+  
+  @objc func setGateway(_ url: URL) {
+    self.braveCore.ipfsAPI.nftIpfsGateway = url
+  }
+  
+  @objc func cancel() {
+    navigationController?.popViewController(animated: true)
+  }
+}
+
+fileprivate class URLInputTableViewCell: UITableViewCell, TableViewReusable {
+  
+  // MARK: UX
+  
+  struct UX {
+    static let textViewHeight: CGFloat = 88
+    static let textViewInset: CGFloat = 16
+  }
+  
+  // MARK: Properties
+  
+  var textview = UITextView(frame: .zero)
+  
+  weak var delegate: UITextViewDelegate? {
+    didSet {
+      textview.delegate = delegate
+    }
+  }
+  // MARK: Lifecycle
+  
+  override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+    super.init(style: style, reuseIdentifier: reuseIdentifier)
+    
+    setup()
+  }
+  
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
+  // MARK: Internal
+  
+  private func setup() {
+    textview = UITextView(frame: CGRect(x: 0, y: 0, width: contentView.frame.width, height: contentView.frame.height)).then {
+      $0.text = "https://"
```

Comment:
i think the default text for this `textView` should be the current NFT gateway value `braveCore.ipfsAPI.nftIpfsGateway?.absoluteString`
we can still keep the `https://` part uneditable.